src/main/java/org/deepsymmetry/beatlink/AnnouncementSocketConnection.java

```diff
@@ -0,0 +1,294 @@
+        return this;
+    }
+
+
+    /**
+     * Keeps track of the registered device announcement listeners.
+     */
+    private final Set<AnnouncementListener> deviceListeners =
+            Collections.newSetFromMap(new ConcurrentHashMap<AnnouncementListener, Boolean>());
+
+    /**
+     * Adds the specified device announcement listener to receive device announcements when DJ Link devices
+     * are found on or leave the network. If {@code listener} is {@code null} or already present in the list
+     * of registered listeners, no exception is thrown and no action is performed.
+     *
+     * <p>Device announcements are delivered to listeners on the
+     * <a href="https://docs.oracle.com/javase/tutorial/uiswing/concurrency/dispatch.html">Event Dispatch thread</a>,
+     * so it is fine to interact with user interface objects in listener methods. Any code in the listener method
+     * must finish quickly, or unhandled events will back up and the user interface will be come unresponsive.</p>
+     *
+     * @param listener the device announcement listener to add
+     */
+    public void addDeviceAnnouncementListener(AnnouncementListener listener) {
+        if (listener != null) {
+            deviceListeners.add(listener);
+        }
+    }
+
+    /**
+     * Removes the specified device announcement listener so that it no longer receives device announcements when
+     * DJ Link devices are found on or leave the network. If {@code listener} is {@code null} or not present
+     * in the list of registered listeners, no exception is thrown and no action is performed.
+     *
+     * @param listener the device announcement listener to remove
+     */
+    public void removeDeviceAnnouncementListener(AnnouncementListener listener) {
+        if (listener != null) {
+            deviceListeners.remove(listener);
+        }
+    }
+
+    /**
+     * Get the set of device announcement listeners that are currently registered.
+     *
+     * @return the currently registered device announcement listeners
+     */
+    @SuppressWarnings("WeakerAccess")
+    public Set<AnnouncementListener> getDeviceAnnouncementListeners() {
+        // Make a copy so callers get an immutable snapshot of the current state.
+        return Collections.unmodifiableSet(new HashSet<AnnouncementListener>(deviceListeners));
+    }
+
+    /**
+     * Stop listening for device announcements. Also discard any announcements which had been received, and
+     * notify any registered listeners that those devices have been lost.
+     */
+    @SuppressWarnings("WeakerAccess")
+    public synchronized void stopAnnouncementSocket() {
+        if (isRunning()) {
+            announcementSocket.get().close();
+            announcementSocket.set(null);
+            flushAnnouncementSocket();
+            deliverLifecycleAnnouncement(logger, false);
+        }
+    }
+
+    public synchronized void flushAnnouncementSocket() {
+        DeviceFinder.getInstance().flush();
```

Comment:
What’s the purpose of this?

Reply:
Once I got down to that class, I realized that `flush()` was just a confusing name, but I think that there is even less reason for a public method like this to exist, and the `flush()` method in DeviceFinder is almost certainly going away too once we bring back its `start()` and `stop()` methods, as discussed below.

Reply:
This is gone, not needed.